src/openforms/submissions/logic/actions.py

```diff
@@ -245,13 +248,17 @@ def eval(
             for item in self.input_mapping
         }
 
-        # Perform DMN call
-        dmn_outputs = evaluate_dmn(
-            definition_id=self.decision_definition_id,
-            version=self.decision_definition_version,
-            input_values=dmn_inputs,
-            plugin_id=self.plugin_id,
-        )
+        def _evaluate_dmn():
+            return evaluate_dmn(
+                definition_id=self.decision_definition_id,
+                version=self.decision_definition_version,
+                input_values=dmn_inputs,
+                plugin_id=self.plugin_id,
+            )
+
+        # Perform DMN call or retrieve result from cache
+        cache_key = hash(str(submission.uuid) + json.dumps(dmn_inputs, sort_keys=True))
```

Comment:
How do we guarantee that the inputs are JSON serializable? What if there ends up a UUID or datetime in the inputs?

Reply:
I'd also expect the definition identifiers to be part of the key - a different table with the same inputs does not produce the same result.

Submission ID then is not strictly necessary, though it's probably a good defensive approach to avoid mixing data of different users

Reply:
> How do we guarantee that the inputs are JSON serializable? What if there ends up a UUID or datetime in the inputs?

I added a similar thing to what the camunda package does, but I realised that this will not work if we have non-serializable types nested in lists/dicts :thinking: Possibly a good idea to update both so that some recursive serialisation is done?